core/templates/dev/head/components/ck-editor-helpers/ck-editor-5-rte.directive.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for CK Editor 5.
+ */
+
+require('services/ContextService.ts');
+require('services/RteHelperService.ts');
+
+const ClassicEditor = require(
+  '@ckeditor/ckeditor5-build-classic/build/ckeditor.js');
+
+var oppia = require('AppInit.ts').module;
+
+oppia.directive('ckEditor5Rte', [
+  'ContextService', 'RteHelperService', 'PAGE_CONTEXT',
+  function(ContextService, RteHelperService, PAGE_CONTEXT) {
+    return {
+      restrict: 'E',
+      scope: {
+        uiConfig: '&'
+      },
+      template: '<div><div></div>' +
```

Comment:
Why do you need the extra div at the start? Add a comment explaining why.

Also, structure this a bit more clearly:

```
 '<div>' +
 '  <div></div>' +
 '  <div ...></div>' +
 '</div>`
```

Reply:
Hi @seanlip, the extra div is needed for the toolbar the editor structure is something like this
```
<div> -> parent enclosure
   <div></div> -> toolbar
   <div></div> -> for editor textarea
</div>
```
I've copied the same template what we were using in CK4 because the structure is justifing and custom styleing can be applied to the toolbar through this structure if needed.

Below picture for reference :
<img width="654" alt="Screenshot 2019-07-27 at 11 54 51 PM" src="https://user-images.githubusercontent.com/41017303/61998189-f5c8bf80-b0c9-11e9-857f-626d7300b71b.png">

Reply:
I see -- consider adding a comment inline to explain. It does help to understand that the first div is for the toolbar and the second is for the editor, but that's not clear from just looking at the template.

Reply:
Done as Suggested Thanks!